### PR TITLE
Show Space IDs in "juju spaces" Output

### PR DESCRIPTION
**Pull request by manadart**
## Description of change

This patch ensures that the output of `juju spaces` shows the new space IDs.

The structure of the command output is changed accordingly, from a map of space name to space data, to a slice of spaces.

## QA steps

- Bootstrap to AWS.
- `juju add-space new-space 172.31.0.0/20`.
- `juju spaces`.
- Output should be:
```
Space  Name       Subnets
0      (default)
1      new-space  252.0.0.0/12
                  172.31.0.0/20
```

## Documentation changes

None.

## Bug reference

N/A

### CODE REVIEW

**Comment by manadart** [comment]
!!build!!

**Comment by manadart** [comment]
[update juju spaces output with the new default space](https://trello.com/c/05hAvKhw/1645-update-juju-spaces-output-with-the-new-default-space)

File: cmd/juju/space/list.go [hunk 1]
```diff
@@ -33,11 +35,19 @@ type ListCommand struct {
...
+The --output argument allows the command's output to be redirected to a file. 
+
+Examples:
+	juju spaces
+	juju spaces --short
+	juju spaces --format yaml
+	juju spaces --format json
+
+See also:
+	add-space
```
[the remark this answers is not in the document]
Comment (hmlanigan) [reply cmd/juju/space/list.go]: added

File: cmd/juju/space/list.go [hunk 2]
```diff
@@ -33,11 +35,19 @@ type ListCommand struct {
...
-their subnets are displayed, otherwise just a list of spaces. The
---format argument has the same semantics as in other CLI commands -
-"yaml" is the default. The --output argument allows the command
-output to be redirected to a file. `
+Displays all defined spaces. By default both spaces and their subnets are displayed.
+Supplying the --short option will list just the space names.
+The --output argument allows the command's output to be redirected to a file. 
+
+Examples:
+	juju spaces
```
[the remark this answers is not in the document]
Comment (hmlanigan) [reply cmd/juju/space/list.go]: done

**Comment by hmlanigan** [comment]
$$merge$$

**Comment by hmlanigan** [comment]
unrelated intermittent test failures

$$merge$$

**Comment by manadart** [comment]
Thanks for shepherding this one home.

**Pull request merged**